.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,21 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+permissions:
+  contents: read
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
```

Comment:
should we use 1.19? I think all other workflows are at 1.19

Reply:
done!